app/code/Magento/Catalog/Block/Product/View/GalleryOptions.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Catalog\Block\Product\View;
+
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Catalog\Block\Product\Context;
+use Magento\Framework\Stdlib\ArrayUtils;
+
+class GalleryOptions extends AbstractView implements ArgumentInterface
+{
+    /**
+     * @var Json
+     */
+    private $jsonSerializer;
+
+    /**
+     * @var Gallery
+     */
+    private $gallery;
+
+    /**
+     * @param Context $context
+     * @param ArrayUtils $arrayUtils
+     * @param Json $jsonSerializer
+     * @param Gallery $gallery
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        ArrayUtils $arrayUtils,
+        Json $jsonSerializer,
+        Gallery $gallery,
+        array $data = []
+    ) {
+        $this->gallery = $gallery;
+        $this->jsonSerializer = $jsonSerializer;
+        parent::__construct($context, $arrayUtils, $data);
+    }
+
+    /**
+     * Retrieve gallery options in JSON format
+     *
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function getOptionsJson()
+    {
+        $optionItems = null;
+
+        //Special case for gallery/nav which can be the string "thumbs/false/dots"
+        is_bool($this->getVar("gallery/nav")) ?
+            $optionItems['nav'] = $this->getVar("gallery/nav") ? 'true' : 'false' :
+            $optionItems['nav'] = $this->escapeHtml($this->getVar("gallery/nav"));
```

Comment:
Such use of ternary operator is quite counter-intuitive. Having `if/else` was much cleaner, even better variant would be to move `$optionItems['nav']` to left side.